src/main/java/com/example/auctionapp/Util/MappingUtility.java

```diff
@@ -0,0 +1,126 @@
+package com.example.auctionapp.Util;
+
+import com.example.auctionapp.dto.BidDto;
+import com.example.auctionapp.dto.CategoryDto;
+import com.example.auctionapp.dto.ImageDto;
+import com.example.auctionapp.dto.ProductDto;
+import com.example.auctionapp.dto.SubcategoryDto;
+import com.example.auctionapp.dto.UserDto;
+import com.example.auctionapp.model.Bid;
+import com.example.auctionapp.model.Category;
+import com.example.auctionapp.model.Image;
+import com.example.auctionapp.model.Product;
+import com.example.auctionapp.model.Subcategory;
+import com.example.auctionapp.model.User;
+
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MappingUtility {
+
+    public static BidDto mapBidToBidDto(Bid bid) {
+
+        return new BidDto(
+                bid.getId(),
+                bid.getDateCreated(),
+                bid.getLastModifiedDate(),
+                bid.getUser().getId(),
+                bid.getUser().getFirstName() + " " + bid.getUser().getLastName(),
+                bid.getUser().getImage().getUrl(),
+                bid.getProduct().getId(),
+                bid.getProduct().getName(),
+                bid.getBidTime().toInstant(ZoneOffset.UTC).toEpochMilli(),
+                bid.getBidAmount()
+        );
+    }
+
+    public static ProductDto mapProductToProductDto(Product product) {
+
+        List<String> images = product.getImages().stream().map(
+                image -> {return image.getUrl();
+                }
+        ).collect(Collectors.toList());
+
+        List<BidDto> bids = new ArrayList<>();
+        if(product.getBids()!=null) {
+            bids = product.getBids().stream().map(
+                    bid -> {
+                        return mapBidToBidDto(bid);
+                    }
+            ).collect(Collectors.toList());
+        }
+        return new ProductDto(product.getId(),
+                product.getDateCreated(),
+                product.getLastModifiedDate(),
+                product.getName(),
+                product.getDescription(),
+                product.getPrice(),
+                product.getSubcategory().getId(),
+                product.getAuctionStartDate().toInstant(ZoneOffset.UTC).toEpochMilli(),
+                product.getAuctionEndDate().toInstant(ZoneOffset.UTC).toEpochMilli(),
```

Comment:
This conversion can be a Utility class. `TimeUtility` or something with a method `.timestampToDateTime` and vice versa